pkg/querier/iterators/chunk_merge_iterator.go

```diff
@@ -78,59 +86,82 @@ outer:
 	return its
 }
 
-func (c *chunkMergeIterator) Seek(t int64) bool {
+func (c *chunkMergeIterator) Seek(t int64) chunkenc.ValueType {
 	c.h = c.h[:0]
 
 	for _, iter := range c.its {
-		if iter.Seek(t) {
+		valType := iter.Seek(t)
+		// TODO for native histograms: allow all types
+		if valType == chunkenc.ValFloat {
 			c.h = append(c.h, iter)
 			continue
 		}
+		if valType != chunkenc.ValNone {
+			c.currErr = fmt.Errorf("unsupported value type %v", valType)
+			return chunkenc.ValNone
+		}
 
 		if err := iter.Err(); err != nil {
 			c.currErr = err
-			return false
+			return chunkenc.ValNone
 		}
 	}
 
 	heap.Init(&c.h)
 
 	if len(c.h) > 0 {
 		c.currTime, c.currValue = c.h[0].At()
-		return true
+		return chunkenc.ValFloat
 	}
 
-	return false
+	return chunkenc.ValNone
 }
 
-func (c *chunkMergeIterator) Next() bool {
+func (c *chunkMergeIterator) Next() chunkenc.ValueType {
```

Comment:
We should check`c.currErr`. If it's set we should immediately return `ValNone`.

Reply:
done

Reply:
done